packages/directory-plugin/src/index.js

```diff
@@ -14,11 +14,11 @@ function directoryPlugin(options) {
 		async resolveId(id, importer) {
 			if (!id.startsWith('dir:')) return;
 
-			const resolved = await this.resolve(id.slice(4) + '\0', importer, { skipSelf: true });
+			const resolved = path.resolve(options.cwd || '.', importer, id.slice(4));
```

Comment:
This might break any sort of aliasing users might do:

```
import whatever from "dir:~/foo";
```

To be able to process those paths we'd need to keep calling into `this.resolve()`